modules/spring-config/src/main/java/org/mule/runtime/config/spring/InjectParamsServiceProxy.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.config.spring;
+
+import static java.lang.String.format;
+import static java.lang.reflect.Modifier.isPublic;
+import static java.lang.reflect.Proxy.newProxyInstance;
+import static java.util.Arrays.asList;
+import static org.mule.runtime.api.util.Preconditions.checkArgument;
+import static org.mule.runtime.core.util.ClassUtils.findImplementedInterfaces;
+
+import org.mule.runtime.api.service.Service;
+import org.mule.runtime.container.api.ServiceInvocationHandler;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.registry.IllegalDependencyInjectionException;
+import org.mule.runtime.core.api.registry.RegistrationException;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+/**
+ * Proxies a {@link Service} instance to automatically {@link Inject} parameters for invocations of implementation methods.
+ * 
+ * @since 4.0
+ */
+public class InjectParamsServiceProxy extends ServiceInvocationHandler {
```

Comment:
I think that the class name should make a reference to MuleContext, as that is what it uses to take the values and in the future there could be another proxy that take parameters from other place